new_client/src/auth/Signin.jsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import { Container, Paper, Grid, TextField, Button } from '@material-ui/core';
+import "./Error.css";
+
+export default function Signin() {
+	const [form, updateForm] = React.useState({
+		username: "",
+		password: "",
+	});
+
+	const [errors, updateErrors] = React.useState({
+		username: "",
+		password: "",
+	});
+
+	function isFormValid() {
+		let formIsValid = true;
+		if (!form.username) {
+			formIsValid = false;
+			updateErrors(prevErrors => ({
+				...prevErrors,
+				username: "*Username can't be Empty"
+			}));
+		}
+
+		if (!form.password) {
+			formIsValid = false;
+			updateErrors(prevErrors => ({
+				...prevErrors,
+				password: "*Please enter your password."
+			}));
+		}
+		
+		return formIsValid;
+
+	}
+
+
+	function handleChange(event) {
+		const { name, value } = event.target;
+		updateForm(prevDetails => {
+			return (
+				{
+					...prevDetails,
+					[name]: value
+				}
+			);
+		});
+	}
+
+	function handleClick(event) {
+		if (isFormValid()) {
+			event.preventDefault();
+			fetch("http://localhost:8000/api/login", {
+				method: "POST",
+				mode: "cors",
+				headers: {
+					"Content-Type": "application/json",
+				},
+				body: JSON.stringify(form),
+			})
+				.then((response) => response.json())
+				.catch((err) => {
+					console.log(err);
+				});
+			}
```

Comment:
Tip: Please always use proper indentation. Nearly all editors have the option to "Reformat Code". Use that

Reply:
The tab space was 4 spaces. That caused an indentation problem. Resolved all indentation to 2 tab spaces.